signer/src/stacks/api.rs

```diff
@@ -320,16 +321,87 @@ impl TenureBlocks {
     /// # Note
     ///
     /// The struct doesn't need to contain all the blocks in a tenure.
+    #[cfg(any(test, feature = "testing"))]
     pub fn into_blocks(self) -> Vec<NakamotoBlock> {
         self.blocks
     }
+}
+
+/// A slimmed down [`NakamotoBlockHeader`]
+#[derive(Debug, Clone, PartialEq)]
+pub struct StacksBlockHeader {
+    /// The total number of StacksBlocks and NakamotoBlocks preceding this
+    /// block in this block's history.
+    pub block_height: u64,
+    /// The identifier for a block. It is the hash of the this block's
+    /// header hash and the block's consensus hash.
+    pub block_id: StacksBlockId,
+    /// The index block hash of the immediate parent of this block. This is
+    /// the hash of the parent block's hash and consensus hash.
+    pub parent_block_id: StacksBlockId,
+}
+
+impl From<NakamotoBlockHeader> for StacksBlockHeader {
+    fn from(value: NakamotoBlockHeader) -> Self {
+        StacksBlockHeader {
+            block_height: value.chain_length,
+            block_id: value.block_id(),
+            parent_block_id: value.parent_block_id,
+        }
+    }
+}
+
+impl From<TenureBlocks> for TenureHeaders {
+    fn from(value: TenureBlocks) -> Self {
+        let headers = value
+            .blocks
+            .into_iter()
+            .map(|block| StacksBlockHeader::from(block.header))
+            .collect();
+
+        TenureHeaders {
+            headers,
+            anchor_block_hash: value.anchor_block_hash,
+            anchor_block_height: value.anchor_block_height,
+        }
+    }
+}
+
+/// This struct represents a non-empty subset of the Stacks block headers
+/// that were created during a tenure.
+#[derive(Debug)]
+pub struct TenureHeaders {
```

Comment:
Maybe `TenureBlockHeaders`? I first read this as `a list of headers of all tenures`

Reply:
https://github.com/stacks-sbtc/sbtc/pull/1604/commits/98036af2577459ebc6f87ba1abc86cebf71cb39e